src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -152,6 +152,10 @@ def create(self, bucket, uid, zonegroup=None, placement_target=None):
             raise DashboardException(e, http_status_code=500, component='rgw')
 
     def set(self, bucket, bucket_id, uid):
+        # When linking a non-tenant-user owned bucket to a tenanted user, we need to prefix bucket
+        # name with '/'. e.g. photos -> /photos
+        if '$' in uid and '/' not in bucket:
+            bucket = '/%s' % bucket
```

Comment:
Please do not use [old-style string formatting](https://pyformat.info/#simple).
```suggestion
            bucket = '/{}'.format(bucket)
```

Reply:
Thanks! Fixed.